chainer/utils/conv_nd.py

```diff
@@ -0,0 +1,126 @@
+import itertools
+import numpy
+import six
+
+from chainer import cuda
+from chainer.utils.conv import get_conv_outsize
+from chainer.utils import conv_nd_kernel
+
+
+def as_tuple(x, n):
+    if hasattr(x, '__getitem__'):
+        assert len(x) == n
+        return tuple(x)
+    return (x,) * n
+
+
+def im2col_nd_cpu(img, ksize, stride, pad, pval=0, cover_all=False):
```

Comment:
Can we assume `ksize`, `stride`, and `pad` have same lengths? Or is it users' responsibility?